server/embed/config.go

```diff
@@ -1136,6 +1145,26 @@ func (cfg *Config) InitialClusterFromName(name string) (ret string) {
 	return ret[1:]
 }
 
+// InferLocalAddr tries to determine the LocalAddr used when communicating with
+// an etcd peer. If SetMemberLocalAddr is true, then it will try to get the host
+// from AdvertisePeerUrls by searching for the first URL with a specified
+// non-loopback address. Otherwise, it defaults to empty string and the
+// LocalAddr used will be the default for the Golang HTTP client.
+func (cfg *Config) InferLocalAddr() string {
+	if !cfg.SetMemberLocalAddr {
+		return ""
+	}
+	for _, peerUrl := range cfg.AdvertisePeerUrls {
+		if addr, err := netip.ParseAddr(peerUrl.Hostname()); err == nil {
+			if addr.IsLoopback() || addr.IsUnspecified() {
+				continue
+			}
+			return addr.String()
+		}
+	}
+	return ""
```

Comment:
Suggest to return error or print warning log here, because user wants to use target IP

Reply:
That's a good point. I added more logging around this area.